jobs: TestRetriesWithExponentialBackoff failed

jobs.TestRetriesWithExponentialBackoff [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12973068?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12973068?buildTab=artifacts#/) on release-23.2 @ [afce36d20a0585fa16cd589322db3884fd5dc9ba](https://github.com/cockroachdb/cockroach/commits/afce36d20a0585fa16cd589322db3884fd5dc9ba):

Fatal error:

```
panic: test timed out after 14m55s
running tests:
	TestRetriesWithExponentialBackoff (14m50s)
	TestRetriesWithExponentialBackoff/revert_on_cancel (14m32s)

```
Stack: 

```
goroutine 604298 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x31c
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestRetriesWithExponentialBackoff
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/b9d280d345e5422f48684ad57ae6f103/logTestRetriesWithExponentialBackoff1161422484
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestRetriesWithExponentialBackoff/revert_on_fail
    test_server_shim.go:146: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
=== RUN   TestRetriesWithExponentialBackoff/running
    test_server_shim.go:146: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
=== RUN   TestRetriesWithExponentialBackoff/pause_running
    test_server_shim.go:146: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
=== RUN   TestRetriesWithExponentialBackoff/revert_on_cancel
    test_server_shim.go:146: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/jobs @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRetriesWithExponentialBackoff.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34095

jobs.TestRetriesWithExponentialBackoff [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13654643?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13654643?buildTab=artifacts#/) on release-23.2 @ [2c32ba829c3dd31ea4c34d3d1eb4a2a54ed4390e](https://github.com/cockroachdb/cockroach/commits/2c32ba829c3dd31ea4c34d3d1eb4a2a54ed4390e):

Fatal error:

```
panic: test timed out after 14m55s
running tests:
	TestRetriesWithExponentialBackoff (14m49s)
	TestRetriesWithExponentialBackoff/revert_on_cancel (14m26s)

```
Stack: 

```
goroutine 918698 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x320
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestRetriesWithExponentialBackoff
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/b9d280d345e5422f48684ad57ae6f103/logTestRetriesWithExponentialBackoff1996051880
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestRetriesWithExponentialBackoff/pause_running
    test_server_shim.go:146: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
=== RUN   TestRetriesWithExponentialBackoff/revert_on_cancel
    test_server_shim.go:146: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRetriesWithExponentialBackoff.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/jobs/registry_test.go
+++ b/pkg/jobs/registry_test.go
@@ -992,7 +992,6 @@ func TestRetriesWithExponentialBackoff(t *testing.T) {
 		bti.clock.AdvanceTo(lastRun)
 		<-bti.resumeCh
 		pauseOrCancelJob(t, ctx, bti.idb, bti.registry, jobID, cancel)
-		bti.errCh <- nil
 		<-bti.failOrCancelCh
 		bti.errCh <- MarkAsRetryJobError(errors.New("injecting error in reverting state"))
 		expectedResumed := bti.resumed.Count()

